File: Qytetet Java/src/modeloqytetet/Tablero.java

package modeloqytetet;
import java.util.ArrayList;
/**
 *
 * @author miguel
 */
public class Tablero {
    private ArrayList<Casilla> casillas ;
    private Casilla carcel;
    
    public Tablero(){
        inicializar();
        carcel=casillas.get(5);

    }
    
    public Tablero(ArrayList<Casilla> casillas, Casilla carcel){
        this.casillas=casillas;
        this.carcel=carcel;
    }
    private void inicializar(){
        casillas = new ArrayList<>();
        int contador=0;
        //Inicializamos primero los títulos de propiedad en un vector auxiliar
        ArrayList<TituloPropiedad> titulos = new ArrayList<>();
        titulos.add(new TituloPropiedad ("Calle Morgana",200,50,7,100,100));
        titulos.add(new TituloPropiedad ("Calle Marcos Heredia",300,75,2,125,125));
        titulos.add(new TituloPropiedad ("Calle Nueva",200,50,7,100,100));
        titulos.add(new TituloPropiedad ("Calle Capo",200,50,7,100,100));
        titulos.add(new TituloPropiedad ("Calle Maxima",200,50,7,100,100));
        titulos.add(new TituloPropiedad ("Calle Migue99angel",200,50,7,100,100));
        titulos.add(new TituloPropiedad ("Calle Estufu",200,50,7,100,100));
        titulos.add(new TituloPropiedad ("Calle Ancha",200,50,7,100,100));
        titulos.add(new TituloPropiedad ("Calle Smite",200,50,7,100,100));
        titulos.add(new TituloPropiedad ("Calle Ela",200,50,7,100,100));
        titulos.add(new TituloPropiedad ("Calle David Broncano",200,50,7,100,100));
        titulos.add(new TituloPropiedad ("Calle calle",200,50,7,100,100));
        
      //Ahora inicializamos el vector casillas
      
        casillas.add(new OtraCasilla (0,TipoCasilla.SALIDA));
        casillas.add(new Calle(1,titulos.get(contador++)));
        casillas.add(new OtraCasilla(2,TipoCasilla.SORPRESA));
        casillas.add(new Calle(3,titulos.get(contador++)));
        casillas.add(new Calle(4,titulos.get(contador++)));
        casillas.add(new OtraCasilla(5,TipoCasilla.CARCEL));
        casillas.add(new Calle(6,titulos.get(contador++)));
        casillas.add(new OtraCasilla(7,TipoCasilla.SORPRESA));
        casillas.add(new Calle(8,titulos.get(contador++)));
        casillas.add(new Calle(9,titulos.get(contador++)));
        casillas.add(new OtraCasilla(10,TipoCasilla.PARKING));
        casillas.add(new Calle(11,titulos.get(contador++)));
        casillas.add(new OtraCasilla(12,TipoCasilla.SORPRESA));
        casillas.add(new Calle(13,titulos.get(contador++)));
        casillas.add(new Calle(14,titulos.get(contador++)));
        casillas.add(new OtraCasilla(15,TipoCasilla.JUEZ));
        casillas.add(new Calle(16,titulos.get(contador++)));
        casillas.add(new OtraCasilla(17,TipoCasilla.IMPUESTO));
        casillas.add(new Calle(18,titulos.get(contador++)));
        casillas.add(new Calle(19,titulos.get(contador++)));
            
    }
    
    ArrayList<Casilla> getCasillas() {
        return casillas;
    }

        Casilla getCarcel() {
            return carcel;
    }
    
    boolean esCasillaCarcel(int numeroCasilla){
        boolean coinciden=false;
        if(numeroCasilla==carcel.getNumeroCasilla())
            coinciden = true;
        return coinciden;
    }
    Casilla obtenerCasillaFinal(Casilla casilla, int desplazamiento){        
        return casillas.get((casilla.getNumeroCasilla()+desplazamiento)%20);
    }
    Casilla obtenerCasillaNumero(int numeroCasilla){
        return casillas.get(numeroCasilla);
    }

    @Override
    public String toString() {
        return "Tablero{" + "casillas=" + casillas + ", carcel=" + carcel + '}';
    }
    
    
    
}

File: Qytetet Ruby/pkg/QytetetRuby-0.0.1/lib/tablero.rb
#encoding: utf-8

require_relative "titulo_propiedad"
require_relative "casilla"
require_relative "tipo_casilla"
require_relative "calle"
module ModeloQytetet
  class Tablero
  

    def initialize
      inicializar_tablero
    
    end

    def self.new_casilla(casillas, carcel)
      @casillas=casillas
      @carcel=carcel
    end

    attr_reader :casillas, :carcel
    
    def to_s
      "Tablero{casillas: #{@casillas},\n carcel: #{@carcel}}"
    end
    
    def inicializar_tablero
      titulos=Array.new
      @casillas=Array.new
      titulos << TituloPropiedad.new("Calle Morgana",750,100,7,100,300)
      titulos << TituloPropiedad.new("Calle Marcos Heredia",800,125,2,125,375)
      titulos << TituloPropiedad.new("Calle Nueva",825,200,7,100,450)
      titulos << TituloPropiedad.new("Calle Capo",975,250,7,100,550)
      titulos << TituloPropiedad.new("Calle Maxima",1000,300,7,100,650)
      titulos << TituloPropiedad.new("Calle Miguel Posadas",1150,350,7,100,700)
      titulos << TituloPropiedad.new("Calle Estufu",1200,375,7,100,850)
      titulos << TituloPropiedad.new("Calle Ancha",1350,400,7,100,875)
      titulos << TituloPropiedad.new("Calle Smite",1400,425,7,100,900)
      titulos << TituloPropiedad.new("Calle Ela",1550,450,7,100,925)
      titulos << TituloPropiedad.new("Calle David Broncano",1600,500,7,100,950)
      titulos << TituloPropiedad.new("Calle calle",1800,600,7,100,1000)

      @casillas << Casilla.new(0, TipoCasilla::SALIDA, 1000)
      @casillas << Calle.new(1, titulos[0])
      @casillas << Casilla.new(2, TipoCasilla::SORPRESA, 0)
      @casillas << Calle.new(3, titulos[1])
      @casillas << Calle.new(4, titulos[2])
      @casillas << Casilla.new(5, TipoCasilla::CARCEL,0)
      @casillas << Calle.new(6, titulos[3])
      @casillas << Casilla.new(7, TipoCasilla::SORPRESA,0)
      @casillas << Calle.new(8, titulos[4])
      @casillas << Calle.new(9, titulos[5])
      @casillas << Casilla.new(10, TipoCasilla::PARKING,0)
      @casillas << Calle.new(11, titulos[6])
      @casillas << Casilla.new(12, TipoCasilla::SORPRESA,0)
      @casillas << Calle.new(13, titulos[7])
      @casillas << Calle.new(14, titulos[8])
      @casillas << Casilla.new(15, TipoCasilla::JUEZ,0)
      @casillas << Calle.new(16, titulos[9])
      @casillas << Casilla.new(17, TipoCasilla::IMPUESTO,0)
      @casillas << Calle.new(18, titulos[10])
      @casillas << Calle.new(19, titulos[11])
      @carcel=@casillas[5]
    end
      private :inicializar_tablero
    def es_casilla_carcel(numeroCasilla)
      es_carcel=false;
      if(@carcel.numeroCasilla==numeroCasilla)
        es_carcel=true;
      return es_carcel;  
      end
    end
    def obtener_casilla_final(casilla,desplazamiento)
      return @casillas[(casilla.numeroCasilla+desplazamiento)%20]
    end
    def obtener_casilla_numero(numeroCasilla)
      return @casillas[numeroCasilla];
    end
  end
  
end